scripts/trainer.py

```diff
@@ -59,7 +59,11 @@ def go(self, args):
 
         model_file_name = f"{model_name}model"
         assert os.path.exists(model_file_name)
-        zstd_compress(model_file_name)
+
+        if os.path.isdir(model_file_name):
+            create_tar_zst(f"{model_file_name}.tar.zst")
+        else:
+            zstd_compress(model_file_name)
```

Comment:
How do we handle this in the decompress process?

Reply:
The decompression process doesn't seem to handle this yet. We could include a similar check to verify if `model_file_name` is a directory and then use the `extract_tar_zst` function for the decompression.

https://github.com/mozilla/bugbug/blob/fc6e17b2d4934c3d3c6ac9a51f91b80f325c2be9/bugbug/utils.py#L217-L233

Reply:
To have the data saved, the compressed file name should match what it is declared in `data-pipeline.yml`.

Example:
https://github.com/mozilla/bugbug/blob/d13c4b15053ec7fdbc0b6ea44acb762d59e315b6/infra/data-pipeline.yml#L983-L991

Reply:
I attempted to compress the directory and give it a name that matches the expected file name, which is "model_file_name.zst." However, I was unsuccessful because I couldn't modify the name. So, instead, I compressed the model files in the directory and renamed the directory to match the expected file name. During the decompression process, the model files within the directory will be decompressed, and the directory will be renamed to remove the ".zst" extension.

https://github.com/mozilla/bugbug/blob/9001ce14cf75fd8cbab21ee3d84f4441b2e97da1/scripts/trainer.py#L63-L74

https://github.com/mozilla/bugbug/blob/9001ce14cf75fd8cbab21ee3d84f4441b2e97da1/bugbug/utils.py#L238-L250

Reply:
Hello @suhaibmujahid, thank you for the review. I've made some changes. I archived the directory and compressed it with a name that matches the expected file name. During decompression, it checks if the resulting file is a `tarfile`; if it is, it extracts the tar archive.

https://github.com/mozilla/bugbug/blob/7a7e399b66c0d5661c90474951024f525b868b45/scripts/trainer.py#L64-L77

https://github.com/mozilla/bugbug/blob/7a7e399b66c0d5661c90474951024f525b868b45/bugbug/utils.py#L237-L245